src/publish/mod.rs

```diff
@@ -210,21 +211,39 @@ pub fn publish_all_contracts(
                 std::process::exit(1);
             }
         }
-        (session.settings, chain, project_manifest)
-    } else {
-        let (settings, chain, project_manifest) =
-            match load_session_settings(manifest_path, &network) {
-                Ok((session, chain, project_manifest)) => (session, chain, project_manifest),
-                Err(e) => return Err(vec![e]),
-            };
-        (settings, chain, project_manifest)
-    };
+    }
 
     let (tx, rx) = channel();
     let (per_contract_event_tx, per_contract_event_rx) = channel();
 
-    let contracts_to_deploy = settings.initial_contracts.len();
-    let node_url = settings.node.clone();
+    let mut contract_asts = HashMap::new();
+    let mut contracts = HashMap::new();
+    for initial_contract in session.settings.initial_contracts.iter() {
+        let contract_id = initial_contract.get_contract_identifier(false).unwrap();
+        contracts.insert(contract_id.clone(), initial_contract);
+        let (ast, _, _) =
+            session
+                .interpreter
+                .build_ast(contract_id.clone(), initial_contract.code.clone(), 2);
+        contract_asts.insert(contract_id.clone(), ast);
+    }
+
+    let result = ASTDependencyDetector::detect_dependencies(&contract_asts, &BTreeMap::new());
+    let dependencies = match result {
+        Ok(deps) => deps,
+        Err(_) => {
+            println!("Deployments in presence of requirements is not supported yet.");
```

Comment:
You should still be able to deploy to mainnet in the presence of requirements, just not to devnet, right? Although since this will be fixed shortly with #333, maybe we can leave this simplification.

Reply:
Could we check here if `network` is `Devnet` and only report this error and exit in that case? Or am I missing something?

Reply:
I think even a Testnet / Mainnet deployment, that involves some requirements would require a special effort for handling the dependencies checking, no?